### DISCUSSION TITLE
not available devices for cooms reasosns

### DISCUSSION THREAD

**Discussion by mariorl** [frangoteam/FUXA#997] [Ideas]
Sometimes is very useful to mark a variable as not available for cooms reasosns, remote devices powered off etc.

Today I deal with that including an incremental word form every device, and put in the graphs as a watchdog,
so that if its value has not changue then the device is not available, and thus every trnd trace from that controller 
is not taken into account on that period. But it is a "manual" deal of comms failure.

Any idea about how to improve that scenario?

thanks, Mario.

**Comment by unocelli** [comment]
Hi,
if you look in FUXA Server Tags you will find a Tag for every connection '[Device name]Connection Status' that should be update at each reading of a value:

```
 var ConnectionStatusEnum = {
    OFF: 0,
    WARNING: 3, // up to 5 times the polling interval without response
    ON: 5,
} 
```

with this logic

```
  if (lastRead < now - pollingInterval * 5) {
      connectionStatus = ConnectionStatusEnum.OFF;
  } else if (lastRead < now - pollingInterval * 2) {
      connectionStatus = ConnectionStatusEnum.WARNING;
  } else {
      connectionStatus = ConnectionStatusEnum.ON;
  }
```
